app/src/main/java/fr/neamar/kiss/result/TagDummyResult.java

```diff
@@ -0,0 +1,140 @@
+        if (mDrawable != null)
+            return mDrawable;
+
+        boolean largeSearchBar = PreferenceManager.getDefaultSharedPreferences(context).getBoolean("large-search-bar", false);
+        int barSize = context.getResources().getDimensionPixelSize(largeSearchBar ? R.dimen.large_bar_height : R.dimen.bar_height);
+
+        int width, height = width = barSize;
+
+        // create a canvas from a bitmap
+        Bitmap bitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
+        Canvas canvas = new Canvas(bitmap);
+
+        // use StaticLayout to draw the text centered
+        TextPaint paint = new TextPaint();
+        paint.setFlags(Paint.ANTI_ALIAS_FLAG);
+        paint.setTextSize(.6f * height);
+
+        RectF rectF = new RectF(0, 0, width, height);
+        rectF.inset(1.f, 1.f);
+
+        // draw a white rounded background
+        paint.setColor(0xFFffffff);
+        canvas.drawRoundRect(rectF, width / 2.4f, height / 2.4f, paint);
+
+        int codepoint = pojo.getName().codePointAt(0);
+        String glyph = new String(Character.toChars(codepoint));
+        // If the codepoint glyph is an image we can't use SRC_IN to draw it.
+        boolean drawAsHole = true;
+        Character.UnicodeBlock block = null;
+        try {
+            block = Character.UnicodeBlock.of(codepoint);
+        } catch (IllegalArgumentException ignored) {
+        }
+        if (block == null)
+            drawAsHole = false;
+        else if ("DINGBATS".equals(block.toString()))
+            drawAsHole = false;
+        else if ("EMOTICONS".equals(block.toString()))
+            drawAsHole = false;
+        else if ("MISCELLANEOUS_SYMBOLS".equals(block.toString()))
+            drawAsHole = false;
+        else if ("MISCELLANEOUS_SYMBOLS_AND_PICTOGRAPHS".equals(block.toString()))
+            drawAsHole = false;
+        else if ("SUPPLEMENTAL_SYMBOLS_AND_PICTOGRAPHS".equals(block.toString()))
+            drawAsHole = false;
+        else if ("TRANSPORT_AND_MAP_SYMBOLS".equals(block.toString()))
+            drawAsHole = false;
+        else if (!"BASIC_LATIN".equals(block.toString())) {
+            // log untested glyphs
+            Log.d(TAG, "Codepoint " + codepoint + " with glyph " + glyph + " is in block " + block);
+        }
+        // we can't draw images (emoticons and symbols) using SRC_IN with transparent color, the result is a square
+        if (drawAsHole) {
+            // write text with "transparent" (create a hole in the background)
+            paint.setColor(0);
+            paint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.SRC_IN));
+        } else {
+            paint.setColor(0xFFffffff);
+        }
+
+        // draw the letter in the center
+        Rect b = new Rect();
+        paint.getTextBounds(glyph, 0, glyph.length(), b);
+        canvas.drawText(glyph, 0, glyph.length(), width / 2.f - b.centerX(), height / 2.f - b.centerY(), paint);
+
+//        rectF.set(b);
```

Comment:
dead code?

Reply:
@Neamar it is useful for debug or if you like how it looks, maybe add a toggle for it.

Reply:
ok, let's keep it then!